src/utils/guildRole.ts

```diff
@@ -84,3 +85,40 @@ export async function removeGuildRole(details: memberGroupRole, env: env) {
     return INTERNAL_SERVER_ERROR;
   }
 }
+
+export async function getGuildRoles(
+  env: env
+): Promise<Array<GuildRole> | undefined> {
+  const guildDetailsUrl = `${DISCORD_BASE_URL}/guilds/${env.DISCORD_GUILD_ID}`;
+
+  try {
+    const response = await fetch(guildDetailsUrl, {
+      method: "GET",
+      headers: {
+        "Content-Type": "application/json",
+        Authorization: `Bot ${env.DISCORD_TOKEN}`,
+      },
+    });
+
+    if (!response.ok) {
+      return undefined;
+    }
+
+    const guildDetails: GuildDetails = await response.json();
+
+    return guildDetails.roles.map((role) => ({
+      id: role.id,
+      name: role.name,
+    }));
+  } catch (err) {
+    return undefined;
```

Comment:
Instead of returning undefined in the error scenario as well, please return a valid error message.